### ISSUE TITLE
[setup-gcloud] already downloaded gcloud binary not found on second and subsequent run

### ISSUE THREAD

**Issue by jacek-jablonski** [GoogleCloudPlatform/github-actions#84]
### TL;DR
I am running self hosted runner and Google Cloud SDK is NOT installed on system. At first run setup-gcloud downloads GoogleCloud SDK and adds gcloud binary location to PATH. Everything works fine. On second run, gcloud cache is found, but gcloud location is not added to PATH. It results in error.

On Github's runner it runs fine, probably because Google Cloud SDK is installed on their VMs system-wide.

**Expected behavior**
Already cached gcloud is added to PATH and action runs fine.

**Observed behavior**
```
::debug::isExplicit: 286.0.0
::debug::explicit? true
::debug::checking cache: /home/actions/cache/gcloud/286.0.0/x64
::debug::Found tool in cache gcloud 286.0.0 x64
::error::Unable to locate executable file: gcloud. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

### Reproduction

**Action YAML**

```yaml
# Paste your complete GitHub Actions YAML here, removing
# any sensitive values.
```

**Repository**

**Additional information**
Running on self-hosted runner.

**Comment by sethvargo** [comment]
Hey @jacek-jablonski - thank you for opening a detailed issue. Can you provide more information about your custom runner setup? If you have some Terraform or gcloud scripts that you used to create them, that would be super helpful for us to reproduce and fix the issue.

**Comment by jacek-jablonski** [comment]
Hi @sethvargo,
Below you can find Dockerfile I'm using:
```
FROM ubuntu:bionic

ARG GIT_VERSION="2.23.0"
ARG GH_RUNNER_VERSION="2.165.2"
ARG TARGETPLATFORM="x64"

SHELL ["/bin/bash", "-o", "pipefail", "-c"]
# hadolint ignore=DL3003
RUN apt-get update && \
  apt-get install -y --no-install-recommends \
    apt-transport-https \
    ca-certificates \
    gnupg-agent \
    curl \
    git \
    software-properties-common \
    python-openssl \
  && rm -rf /var/lib/apt/lists/*

RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - \
  && [[ $(lsb_release -cs) == "eoan" ]] && ( add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu disco stable" ) || ( add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" )\
  && apt-get update \
  && apt-get install -y docker-ce docker-ce-cli containerd.io --no-install-recommends \
  && rm -rf /var/lib/apt/lists/*

RUN curl -sL "https://github.com/docker/compose/releases/download/1.25.4/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose \
  && chmod +x /usr/local/bin/docker-compose

WORKDIR /actions-runner
COPY install_actions.sh /actions-runner

RUN chmod +x /actions-runner/install_actions.sh \
  && /actions-runner/install_actions.sh ${GH_RUNNER_VERSION} ${TARGETPLATFORM} \
  && rm /actions-runner/install_actions.sh

COPY entrypoint.sh /
RUN chmod +x /entrypoint.sh
ENTRYPOINT ["/entrypoint.sh"]

```
